### PR TITLE
Add prince-factor-analysis recipe

### PR DESCRIPTION
**Pull request by mwilson8**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by mwilson8** [comment]
@conda-forge/bioconda-recipes this is the first time I've seen this lint error. How to proceed?

**Comment by mwilson8** [comment]
re-ping @conda-forge/bioconda-recipes please advise

File: recipes/prince/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  version: "{{ version }}"
+
+source:
+  url: "https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz"
+  sha256: f6dc02fa63bf2b1b0a5edb7c1f30efc062e498bbf705a207b8bb5e1fa3fec4fa
+
+build:
+  number: 0
+  script: "{{ PYTHON }} -m pip install . -vv"
+  skip: True  # [py2k]
```
Comment (marcelotrevisani) [inline recipes/prince/meta.yaml]: I think this recipe can be ``noarch`` and you can restrict the ``python`` version in the ``requirements``
```suggestion
  noarch: python
```

File: recipes/prince/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  
+requirements:
+  host:
+    - matplotlib
+    - numpy
+    - pandas
+    - pip
+    - python
```
Comment (marcelotrevisani) [inline recipes/prince/meta.yaml]:
```suggestion
    - python >=3.0
```

File: recipes/prince/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    - scikit-learn
+    - scipy
+  run:
+    - matplotlib >=3.0.2
+    - numpy >=1.16.1
+    - pandas >=0.24.0
+    - python
```
Comment (marcelotrevisani) [inline recipes/prince/meta.yaml]:
```suggestion
    - python >=3.0
```

File: recipes/prince/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  sha256: f6dc02fa63bf2b1b0a5edb7c1f30efc062e498bbf705a207b8bb5e1fa3fec4fa
+
+build:
+  number: 0
+  script: "{{ PYTHON }} -m pip install . -vv"
+  skip: True  # [py2k]
+  
+requirements:
+  host:
+    - matplotlib
```
Comment (marcelotrevisani) [inline recipes/prince/meta.yaml]: Do you need all these dependencies to install the package or just ``python``, ``setuptools`` and ``pip``?

**Comment by mwilson8** [comment]
@marcelotrevisani suggested changes applied, ready for review

**Comment by mbargull** [comment]
> @conda-forge/bioconda-recipes this is the first time I've seen this lint error. How to proceed?

Hi @mwilson8, this means there already exists a Conda package named `prince`, just not in the `conda-forge` but in the `bioconda` channel on anaconda.org. Both projects, conda-forge and Bioconda, collaborate with each other with `bioconda` being a bioinformatics-centered channel that uses `conda-forge` for its non-bio-base dependencies.

These kinds of package name conflicts sadly happen from time to time, esp. for names that are "short common english word" (like `prince` in this case), since Conda packages are distinguished just by their name.

We do not have strict policies in place, neither in conda-forge nor Bioconda, that dictate which package gets to choose/"claim" such name. Hence, it's more or less a "first come first serve" process (which of course sometimes leads to unfortuante situations, e.g., "less popular" packages having taken names from "more popular" packages).

In any case, to not break existing Conda users' environments, I'd like to ask you to change the name to avoid the clash, e.g., prepending `python-` or pre-/appending some other appropriate affix specific to the origin/nature/purpose of the package (maybe something with "analysis" in this case?).

(Sorry for the inconvenience. I know these things are annoying...)

**Comment by mwilson8** [comment]
@mbargull thanks for the response. I figured it was something along those lines, just wasn't sure how to proceed

> In any case, to not break existing Conda users' environments, I'd like to ask you to change the name to avoid the clash, e.g., prepending `python-` or pre-/appending some other appropriate affix specific to the origin/nature/purpose of the package (maybe something with "analysis" in this case?).

Due to the package in bioconda also having the name [PythonPRINCE](https://github.com/WGS-TB/PythonPrince) on GitHub, I'll change this to something with "analysis" appended as suggested

**Comment by mbargull** [comment]
Thanks, I agree. (Yep, I already facepalmed upon seeing the `PythonPrince` != `prince` oddity...)

**Comment by mwilson8** [comment]
@conda-forge-admin please rerender

**Comment by mwilson8** [comment]
@rluria14 maintain this with me?

**Pull request merged**